src/lando/lando/settings.py

```diff
@@ -10,24 +10,41 @@
 https://docs.djangoproject.com/en/dev/ref/settings/
 """
 
+
+import os
 from pathlib import Path
 
-# Build paths inside the project like this: BASE_DIR / 'subdir'.
-BASE_DIR = Path(__file__).resolve().parent.parent
 
+def get_setting(
+    key: str, default: any, source: any = os.environ, target_type=None
+) -> any:
+    """Fetch a setting from a source defined by `source`.
 
-# Quick-start development settings - unsuitable for production
-# See https://docs.djangoproject.com/en/dev/howto/deployment/checklist/
+    `source` should have a `get` method defined, which will be called when fetching the
```

Comment:
It would be better if we passed the `get` method as the source instead (ie `os.environ.get`) and then the type hint will be `Callable` instead of `Any`.

Reply:
I don't think we should be changing the implementation of something solely to satisfy type hinting -- rather, the other way around would make sense.

I think that passing `os.environ.get` (or, in the case when `source` is say, a dictionary, passing `some_dict.get`) is awkward. In this particular case, we could use `Mapping` as a type hint to improve this type hint.

Reply:
My point was that passing `Any` when the actual constraint is "an object with a `.get`" isn't very helpful, and we should constrain it within the type system where possible. Using a `Mapping` type hint would have worked as well, but I like the simpler solution of `os.getenv`.

Reply:
Yep agreed -- this should have been a `Mapping` type from the get go and `Any` was indeed too broad.